llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -1801,11 +1801,8 @@ void DAGCombiner::Run(CombineLevel AtLevel) {
 
     if (N->getNumValues() == RV->getNumValues())
       DAG.ReplaceAllUsesWith(N, RV.getNode());
-    else {
-      assert(N->getValueType(0) == RV.getValueType() &&
-             N->getNumValues() == 1 && "Type mismatch");
```

Comment:
This can help to catch unexpected nodes. Do we need to remove both?

Reply:
Yes, we need. After changing `SUB ` to `CCMP`, neither of them satisfy.  For this case, `N->getValueType(1) == RV.getValueType() && N->getNumValues() == 2`

Reply:
> Yes, we need. After changing SUB to CCMP, neither of them satisfy.

This means that there is a problem in combining SUB to CCMP. The assert here is to catch such kind of errors.
See my other comment.